README.md

```diff
@@ -292,6 +292,37 @@ The metadata should be written directly to the request body.
 }
 ```
 
+### GET /git/fetch
+
+Does a git fetch on the repository (relevant for when runners clone down git repositories)
+
+### GET /git/branches
+
+Returns an array of branches with their `name`, `commit_hash` (where branch pointer is), `time` (of last commit) and `message` (of last commit)
+
+#### Response
+
+```
+{
+  "status": "success",
+  "data": [
+    {
+      "name": "main",
+      "commit_hash": "ede307e23b2137ba2c7c3270e52f354f224942af",
+      "time": 1722436575,
+      "message": "First commit\n"
+    },
+    {
+      "name": "other",
+      "commit_hash": "e9078cf779584168c3781379380a3b1352545cda",
+      "time": 1722436640,
+      "message": "Second commit\n"
```

Comment:
Will the embedded newlines work well here? I've had mixed success with doing this in json (dim memories of escaping issues), and I wonder if an array of strings might be better?

Reply:
rip ill heed your warning, have changed it to array of strings